specification/hugr.md

```diff
@@ -423,15 +423,86 @@ inputs of successor `i`.
 Some normalizations are possible:
 
   - If the entry node has no predecessors (i.e. is not a loop header),
-    then its contents can be moved outside the ??-node into a containing
+    then its contents can be moved outside the CFG node into a containing
     DSG.
+  - If the entry node has only one successor and that successor is the
+    exit node, the CFG node itself can be removed.
 
-  - If the entry node’s has only one successor and that successor is the
-    exit node, the CFG-node itself can be removed
+The CFG in the example below has three inputs: one (call it `v`) of type "P"
+(not specified, but with a conversion to boolean represented by "P?"), one of
```

Comment:
Note the `P?` nodes in the graph are not doing the same thing - their types are different, as they are putting different values into the tuples (inside the sum). I'd either
1. reword this to suggest that they are different (perhaps even rename - P1 and P2, or Query1 and Query2, say); or,
2. use a simple bool type `[()|()]` everywhere, and therefore the same set of inputs (including both P and angle) to both BB1 and BB3, and then ideally note where the extra Inputs are being discarded. (Since we don't have explicit copy/discard anymore I'm not sure about the last, maybe it'd be ok to silently drop them).